### PR TITLE
Integrate services of all repos to get them working.

### PR DESCRIPTION
**Pull request by nishanbajracharya**
Changes made:

- Added chill-core package
- Renamed `rest-api` to `api`
- Moved migrations into `monitor`
- Used `sqlite3` queries instead of `postgresql` in `api`
- Added `env` config in `api` and `dashboard` to point to `chill.yml` file of `monitor`

### CODE REVIEW

File: api/.env.example [hunk 1]
```diff
@@ -0,0 +1 @@
+CHILL_CONFIG='chill.yml'
```
Comment (kabirbaidhya) [inline api/.env.example]: newline? Update editorconfig?

File: api/package.json [hunk 2]
```diff
@@ -56,6 +57,7 @@
     "babel-jest": "^19.0.0",
     "babel-loader": "^8.0.0",
     "babel-plugin-dynamic-import-node": "^1.0.2",
+    "dotenv": "^6.2.0",
```
Comment (kabirbaidhya) [inline api/package.json]: Actually, we initially tried of getting rid of `.env` file as we had `chill.yml`

File: monitor/src/services/statusLog.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import logger from '../utils/logger';
+import StatusLog from '../models/StatusLog';
+import { fetch as fetchStatus } from './status';
+import { fetch as fetchService } from './service';
+
+/**
+ * Get all status logs.
+ *
+ * @return {Promise}
```
Comment (kabirbaidhya) [inline monitor/src/services/statusLog.js]: `return` -> `returns`. Let's stick to this everywhere.

File: monitor/src/services/status.js [hunk 4]
```diff
@@ -71,3 +73,52 @@ function shouldRetry(err, method) {
...
+  let result = await new Status({ name }).fetch();
+
+  if (!result) {
+    throw new Boom.notFound('Status not found');
+  }
+
+  logger().debug('Retrieved Status data', result.toJSON());
+
+  return result;
+}
```
Comment (kabirbaidhya) [inline monitor/src/services/status.js]: Newline - I think somehow it's not taking editorconfig.

File: monitor/src/services/status.js [hunk 5]
```diff
@@ -71,3 +73,52 @@ function shouldRetry(err, method) {
...
+  logger().info('Fetching all the statuses.');
+
+  let result = await Status.fetchAll();
+
+  logger().debug('Retrieved list of statuses', result.toJSON());
+
+  return result;
+}
+
+export async function fetchByName(name) {
```
Comment (kabirbaidhya) [inline monitor/src/services/status.js]: .

File: monitor/src/services/service.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    const service = await Service.create(data);
+
+    return service.toJSON();
+  } catch (err) {
+    logger().error('Error while persisting the service into database', err);
+  }
+}
+
+/**
+ *
```
Comment (kabirbaidhya) [inline monitor/src/services/service.js]: .

File: api/src/queries/status.js [hunk 7]
```diff
@@ -1,29 +1,30 @@
...
+  json_object(
+    'id', s.id,
+    'name', s.name,
+    'url', s.url,
+    'type', s.type
+  ) as service
   FROM status_logs sl
   INNER JOIN services s ON s.id = sl.service_id
   INNER JOIN statuses st ON st.id = sl.status_id
+  GROUP By sl.service_id
```
Comment (mesaugat) [inline api/src/queries/status.js]: `BY`

File: monitor/src/services/service.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+ * @param {string} url
+ */
```
Comment (kabirbaidhya) [inline monitor/src/services/service.js]: `@returns`

File: core/.editorconfig [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# editorconfig.org
+root = true
+
```
Comment (mesaugat) [inline core/.editorconfig]: Move this out of `core` folder?

File: monitor/src/services/persistence.js [hunk 10]
```diff
@@ -47,3 +91,32 @@ export async function getLastStatus(serviceName) {
...
 }
+
+/**
+ * Compute the diff between the configured and persisted services.
+ *
+ * @param {Array} configuredServices
+ * @param {Array} persistedServices
+ * @returns
+ */
+function computeDiff(configuredServices, persistedServices) {
```
Comment (kabirbaidhya) [inline monitor/src/services/persistence.js]: just `diff` could suffice maybe?

File: monitor/src/services/persistence.js [hunk 11]
```diff
@@ -47,3 +91,32 @@ export async function getLastStatus(serviceName) {
     logger().error('Error while fetching status change', err);
   }
 }
+
+/**
+ * Compute the diff between the configured and persisted services.
+ *
+ * @param {Array} configuredServices
+ * @param {Array} persistedServices
+ * @returns
```
Comment (kabirbaidhya) [inline monitor/src/services/persistence.js]: .

File: dashboard/webpack.config.production.js [hunk 12]
```diff
@@ -1,11 +1,13 @@
 const path = require('path');
 const webpack = require('webpack');
-const { init } = require('chill-core');
+const { init } = require('../core/dist');
```
Comment (kabirbaidhya) [inline dashboard/webpack.config.production.js]: Eventually need to use the published package.

File: dashboard/src/components/dashboard/ServiceRow.js [hunk 13]
```diff
@@ -5,23 +5,27 @@ import ToolTip from 'react-tooltip';
 
 import { isUp, getServiceParams } from '../../services/status';
 
-const ServiceRow = ({ service }) => {
-  let { id, name, createdAt } = service;
-  let { message, className, icon } = getServiceParams(isUp(service));
+const ServiceRow = ({ data }) => {
+  let { id, createdAt, service, status } = data;
```
Comment (kabirbaidhya) [inline dashboard/src/components/dashboard/ServiceRow.js]: const?

File: dashboard/src/components/dashboard/ServiceRow.js [hunk 14]
```diff
@@ -5,23 +5,27 @@ import ToolTip from 'react-tooltip';
...
+
+  const serviceData = JSON.parse(service);
+  const statusData = JSON.parse(status);
+
+  let { message, className, icon } = getServiceParams(isUp(statusData));
```
Comment (kabirbaidhya) [inline dashboard/src/components/dashboard/ServiceRow.js]: const?

**Comment by mesaugat** [comment]
@nishanbajracharya Remove `.editorconfig` from all folders and use a single file at the top level directory. Also, remove all `.codeclimate.yml` files.

File: monitor/src/services/status.js (hunk 5 above)
Comment (mesaugat) [reply monitor/src/services/status.js]: Missing JSDoc.

File: core/chill.yml.dist [hunk 15]
```diff
@@ -1,13 +1,22 @@
 logging:
   level: debug
 
+dashboard:
+  baseHref: '/'
```
Comment (kabirbaidhya) [inline core/chill.yml.dist]: baseUrl

File: core/chill.yml.dist [hunk 16]
```diff
@@ -1,13 +1,22 @@
...
+  logoUrl: LOGO_URL
```
Comment (kabirbaidhya) [inline core/chill.yml.dist]:
```
logo:
  url: LOGO_URL
  height: 80px
```

File: core/src/config/default.config.js [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+    tsFormat: () => new Date().toISOString()
+  },
+  db: {
+    client: 'sqlite3',
+    connection: {
+      filename: null
+    },
+    useNullAsDefault: true
+  },
+  restApi: {
```
Comment (kabirbaidhya) [inline core/src/config/default.config.js]: api

File: api/src/config/default.config.js [hunk 18]
```diff
@@ -24,6 +20,13 @@ export default {
   restApi: {
```
Comment (kabirbaidhya) [inline api/src/config/default.config.js]: api

**Pull request merged**